slowfast/config/defaults.py

```diff
@@ -537,6 +537,13 @@
 _C.FSDP.ENABLED = False
 # Enable resharding after the FW pass. This saves memory but tradesoff communication.
 _C.FSDP.RESHARD_AFTER_FW = False
+# Enable nested wrapping in FSDP
+_C.FSDP.NESTED_WRAP = False
+# USE FSDP auto wrapping
+_C.FSDP.AUTO_WRAP = False
+# Minimum number of params to wrap a module.
+_C.FSDP.MIN_PARAMS_TO_WRAP = 1e3
+
```

Comment:
So I think we should clean this up a bit. Nested wrapping is supposed to support manually specifying wrapping, whereas auto wrap relies on FSDP's auto wrap functionality (which needs min params). In pycls, we use `MIN_PARAMS_TO_WRAP` in nested wrapping which IMO is almost a bug.

cc @lyttonhao, @vaibhava0 do we want the API to support both kinds of wrapping or just one? I personally believe just supporting nested wrap should be enough. Otherwise it's too much maintenance.

Reply:
@mannatsingh That's true, `MIN_PARAMS_TO_WRAP` is now used for `auto_wrap` as well as in the logic of manual wrapping. Do you mean that the name should be changed? or we shouldn't use this parameter when manually wrapping? (because in some sense, it can be useful so that you don't wrap very small layers)

Reply:
> or we shouldn't use this parameter when manually wrapping? (because in some sense, it can be useful so that you don't wrap very small layers)

I meant this. The reason was that almost everyone in pycls spent some time debugging an issue where there FSDP setup changed based on the model size. It was because of this setting. What this setting means is that `nested_wrap` will not wrap any blocks with `FSDP` for models smaller than a certain size. I think when manually asking for wrapping, the users already know that they want wrapping, and this gets in the way. But this was just a suggestion, I'll let the people who will be using this codebase make the final call :)